File: backend/src/models/Together.php
<?php
/**
 * Together Model
 */

namespace MillalHomepage\Models;

use MillalHomepage\Utils\Database;

class Together {
    private $db;
    
    public function __construct() {
        $this->db = Database::getInstance()->getConnection();
    }
    
    /**
     * 함께하는 교회 항목 목록 조회
     */
    public function getAll($onlyActive = true) {
        try {
            if ($onlyActive) {
                $stmt = $this->db->prepare("
                    SELECT * FROM together_items 
                    WHERE is_active = 1 
                    ORDER BY `order` ASC
                ");
            } else {
                $stmt = $this->db->prepare("
                    SELECT * FROM together_items 
                    ORDER BY `order` ASC
                ");
            }
            
            $stmt->execute();
            return $stmt->fetchAll();
        } catch (\PDOException $e) {
            error_log("Get all together items error: " . $e->getMessage());
            return [];
        }
    }
    
    /**
     * 항목 상세 조회
     */
    public function getById($id) {
        try {
            $stmt = $this->db->prepare("SELECT * FROM together_items WHERE id = ?");
            $stmt->execute([$id]);
            
            return $stmt->fetch();
        } catch (\PDOException $e) {
            error_log("Get together item error: " . $e->getMessage());
            return null;
        }
    }
    
    /**
     * 항목 생성
     */
    public function create($data) {
        try {
            $stmt = $this->db->prepare("
                INSERT INTO together_items (
                    title, description, image, link, `order`, is_active
                ) VALUES (?, ?, ?, ?, ?, ?)
            ");
            
            $stmt->execute([
                $data['title'],
                $data['description'] ?? null,
                $data['image'] ?? null,
                $data['link'] ?? null,
                $data['order'] ?? 0,
                $data['isActive'] ?? true
            ]);
            
            return ['success' => true, 'id' => $this->db->lastInsertId()];
        } catch (\PDOException $e) {
            error_log("Create together item error: " . $e->getMessage());
            return ['success' => false, 'error' => 'Failed to create item'];
        }
    }
    
    /**
     * 항목 수정
     */
    public function update($id, $data) {
        try {
            $stmt = $this->db->prepare("
                UPDATE together_items SET 
                    title = ?, 
                    description = ?, 
                    image = ?, 
                    link = ?, 
                    `order` = ?, 
                    is_active = ? 
                WHERE id = ?
            ");
            
            $stmt->execute([
                $data['title'],
                $data['description'] ?? null,
                $data['image'] ?? null,
                $data['link'] ?? null,
                $data['order'] ?? 0,
                $data['isActive'] ?? true,
                $id
            ]);
            
            return ['success' => true];
        } catch (\PDOException $e) {
            error_log("Update together item error: " . $e->getMessage());
            return ['success' => false, 'error' => 'Failed to update item'];
        }
    }
    
    /**
     * 항목 삭제
     */
    public function delete($id) {
        try {
            $stmt = $this->db->prepare("DELETE FROM together_items WHERE id = ?");
            $stmt->execute([$id]);
            
            return ['success' => true];
        } catch (\PDOException $e) {
            error_log("Delete together item error: " . $e->getMessage());
            return ['success' => false, 'error' => 'Failed to delete item'];
        }
    }
}
?>

File: backend/app/routers/together.py
from fastapi import APIRouter, Depends
from pymysql.connections import Connection

from ..auth import require_role
from ..database import get_db, serialize, serialize_all
from ..response import success, error

router = APIRouter(prefix="/together", tags=["together"])


@router.get("")
def get_all(db: Connection = Depends(get_db)):
    with db.cursor() as cur:
        cur.execute(
            "SELECT * FROM together_items WHERE is_active = 1 ORDER BY `order`"
        )
        rows = cur.fetchall()
    return success(serialize_all(rows))


@router.get("/{item_id}")
def get_one(item_id: int, db: Connection = Depends(get_db)):
    with db.cursor() as cur:
        cur.execute("SELECT * FROM together_items WHERE id = %s", (item_id,))
        row = cur.fetchone()
    if not row:
        return error("Not found", "NOT_FOUND", 404)
    return success(serialize(row))


@router.post("")
def create(body: dict, db: Connection = Depends(get_db), _=Depends(require_role("editor"))):
    with db.cursor() as cur:
        cur.execute(
            """INSERT INTO together_items (title, description, image, is_active, `order`)
               VALUES (%s, %s, %s, %s, %s)""",
            (
                body.get("title"),
                body.get("description"),
                body.get("image"),
                body.get("is_active", True),
                body.get("order", 0),
            ),
        )
        new_id = cur.lastrowid
    with db.cursor() as cur:
        cur.execute("SELECT * FROM together_items WHERE id = %s", (new_id,))
        row = cur.fetchone()
    return success(serialize(row), status_code=201)


@router.put("/{item_id}")
def update(item_id: int, body: dict, db: Connection = Depends(get_db), _=Depends(require_role("editor"))):
    with db.cursor() as cur:
        cur.execute("SELECT id FROM together_items WHERE id = %s", (item_id,))
        if not cur.fetchone():
            return error("Not found", "NOT_FOUND", 404)
        cur.execute(
            """UPDATE together_items SET title=%s, description=%s, image=%s,
               is_active=%s, `order`=%s WHERE id=%s""",
            (
                body.get("title"),
                body.get("description"),
                body.get("image"),
                body.get("is_active"),
                body.get("order"),
                item_id,
            ),
        )
    with db.cursor() as cur:
        cur.execute("SELECT * FROM together_items WHERE id = %s", (item_id,))
        row = cur.fetchone()
    return success(serialize(row))


@router.delete("/{item_id}")
def delete(item_id: int, db: Connection = Depends(get_db), _=Depends(require_role("editor"))):
    with db.cursor() as cur:
        cur.execute("SELECT id FROM together_items WHERE id = %s", (item_id,))
        if not cur.fetchone():
            return error("Not found", "NOT_FOUND", 404)
        cur.execute("DELETE FROM together_items WHERE id = %s", (item_id,))
    return success(None, "Deleted successfully")
